src/compiler/scala/tools/nsc/transform/Delambdafy.scala

```diff
@@ -365,7 +365,7 @@ abstract class Delambdafy extends Transform with TypingTransformers with ast.Tre
         else {
           seen += symbol
           (thisReferringMethods contains symbol) ||
-            (liftedMethodReferences(symbol) exists loop) && {
+            (liftedMethodReferences.contains(symbol) && liftedMethodReferences(symbol).exists(loop)) && {
```

Comment:
Could you elaborate on this change?

Reply:
Is there a way to only look up `symbol` once?

Reply:
Yes, we can call `exists` on `liftedMethodReferences`.

Reply:
`exists` can't do a hash lookup, so it will be much slower for large sets. The new implementation performs the hash lookup twice when the symbol is found which is a bit slower than before but it doesn't have to instantiate and traverse a new `mutable.Set` when the symbol is not found which should be considerably faster. Unfortunately I don't know how common these two cases are in a typical codebase. Another option would be to keep a single empty set around as a dummy value that could be used with `getOrElse` (instead of `apply`) or use `get` and live with the boxing overhead.

Reply:
Ok, I think the new version is more clear. If there's a performance problem (seems unlikely), we can fix later.